syntax = "proto3";

message Layer {
    enum LayerType {
        LINEAR = 0;
        RELU = 1;
        LOGSOFTMAX = 2;
    }

    repeated LayerType type = 1;
}

message ModelLayers {
    repeated Layer layers = 1;

    repeated LinearLayer linearLayers = 2;
     
    repeated ReLULayer reluLayers = 3;

    repeated LogSoftmaxLayer logsoftmaxLayers = 4;

    message LinearLayer {
        repeated int32 inFeatures = 1;
        repeated int32 outFeatures = 2;
    }

    message ReLULayer {
        repeated bool inPlace = 1;
    }

    message LogSoftmaxLayer {
        repeated int32 dim = 1;
    }

}

